### PR TITLE
refactor:feat(ecs/volume): add new methods to support disk mount

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
The method provided by the current V1 version of the block device package is not enough to support the research and development needs of disk mounting.

**Which issue this PR fixes**
```
NONE
```

**Special notes for your reviewer**:

**Release note**:

```release-note
1. add new jobs sdk support.
2. refactor block_device sdk methods.
```

### CODE REVIEW

File: openstack/ecs/v1/block_devices/requests.go [hunk 1]
```diff
@@ -2,9 +2,108 @@ package block_devices
...
 
-func Get(c *golangsdk.ServiceClient, server_id string, volume_id string) (r GetResult) {
-	_, r.Err = c.Get(getURL(c, server_id, volume_id), &r.Body, nil)
-	return
+var requestOpts = golangsdk.RequestOpts{
+	MoreHeaders: map[string]string{"Content-Type": "application/json", "X-Language": "en-us"},
+}
+
+// Get is a method to obtain the detail of volume attachment for specified cloud server.
+func Get(c *golangsdk.ServiceClient, serverId string, volumeId string) (*Attachment, error) {
```
Comment (ShiChangkuo) [inline openstack/ecs/v1/block_devices/requests.go:13]: why change the return type?

Comment (Lance52259) [reply openstack/ecs/v1/block_devices/requests.go]: Changes have been reworked to be compatible with legacy code

File: openstack/ecs/v1/block_devices/requests.go [hunk 2]
```diff
@@ -2,9 +2,100 @@ package block_devices
...
+
+// Attach is a method to mount a disk to a specified cloud server.
+func Attach(c *golangsdk.ServiceClient, opts AttachOpts) (*jobs.Job, error) {
+	b, err := golangsdk.BuildRequestBody(opts, "volumeAttachment")
+	if err != nil {
+		return nil, err
+	}
+
+	var rst golangsdk.Result
+	_, err = c.Post(attachURL(c, opts.ServerId), b, &rst.Body, &golangsdk.RequestOpts{
```
Comment (hmbig2) [inline openstack/ecs/v1/block_devices/requests.go:60]: s/&rst.Body/*jobs.Job

File: openstack/ecs/v1/block_devices/requests.go [hunk 3]
```diff
@@ -2,9 +2,100 @@ package block_devices
...
+func Detach(c *golangsdk.ServiceClient, volumeId string, opts DetachOpts) (*jobs.Job, error) {
+	url := detachURL(c, opts.ServerId, volumeId)
+	query, err := golangsdk.BuildQueryString(opts)
+	if err != nil {
+		return nil, err
+	}
+	url += query.String()
+
+	var rst golangsdk.Result
+	_, err = c.DeleteWithBody(url, &rst.Body, &golangsdk.RequestOpts{
```
Comment (hmbig2) [inline openstack/ecs/v1/block_devices/requests.go:92]: s/&rst.Body/*jobs.Job

File: openstack/ecs/v1/block_devices/requests.go (hunk 2 above)
Comment (Lance52259) [reply openstack/ecs/v1/block_devices/requests.go]: done

File: openstack/ecs/v1/block_devices/requests.go (hunk 3 above)
Comment (Lance52259) [reply openstack/ecs/v1/block_devices/requests.go]: done

**Pull request merged**